### PR TITLE
Add swift-atomics

### PR DESCRIPTION
**Pull request by clackary**
### Pull Request Description

Add https://github.com/apple/swift-atomics

### Acceptance Criteria
- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 4.0 and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
    * Apache License, version 2.0
- [x] pass `./project_precommit_check` script run
```
PASS: swift-atomics, 5.2, df9f81, Swift Package
========================================
Action Summary:
     Passed: 1
     Failed: 0
    XFailed: 0
    UPassed: 0
      Total: 1
========================================
Repository Summary:
      Total: 1
========================================
Result: PASS
========================================
--- swift-atomics checked successfully against Swift 5.2 ---
```

### CODE REVIEW

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@lorentey FYI - Adding you as a maintainer for this! Thanks

File: projects.json [hunk 1]
```diff
@@ -2621,6 +2621,32 @@
...
+    "branch": "main",
+    "maintainer": "klorentey@apple.com",
+    "compatibility": [
+      {
+        "version": "5.2",
+        "commit": "df9f811f19a852fa7b78af9cffa8ec447eaeb40a"
+      }
+    ],
+    "platforms": [
+      "Darwin"
```
Comment (lorentey) [inline projects.json:2637]: I believe we can include `"Linux"` as a supported platform here.

**Review by lorentey — APPROVED** [review]
Thanks for adding this! 💯

**Comment by shahmishal** [comment]
@lorentey Have you seen this error before? 

Full log: https://ci.swift.org/job/swift-PR-source-compat-suite-test-macOS/680/artifact/swift-source-compat-suite/FAIL_swift-atomics_5.2_BuildSwiftPackage.log
```
/Users/buildnode/jenkins/workspace/swift-PR-source-compat-suite-test-macOS/swift-source-compat-suite/project_cache/swift-atomics: warning: failed storing manifest for 'swift-atomics' in cache: unknown system error while operating on /Users/buildnode/Library/Caches/org.swift.swiftpm/manifests/manifest.db.lock
```

**Comment by clackary** [comment]
The previous log looks strange to me, there are a few warnings, but no real error after the swift-frontend command.

```
warning: failed storing manifest for 'swift-atomics' in cache: unknown system error while operating on /Users/buildnode/Library/Caches/org.swift.swiftpm/manifests/manifest.db.lock
```

The warning though, makes me wonder if something else caused this to fail. Going to restart testing for now...

**Comment by clackary** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
> The previous log looks strange to me, there are a few warnings, but no real error after the swift-frontend command.
> 
> ```
> warning: failed storing manifest for 'swift-atomics' in cache: unknown system error while operating on /Users/buildnode/Library/Caches/org.swift.swiftpm/manifests/manifest.db.lock
> ```
> 
> The warning though, makes me wonder if something else caused this to fail. Going to restart testing for now...

Looks like the same error I wrote above

File: projects.json (hunk 1 above)
Comment (clackary) [reply projects.json]: Fixed

**Comment by clackary** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@shahmishal I think this should be fine to merge now. The failure in PR testing is coming from swift-futures, which is also failing on the baseline, and swift-atomics built successfully.

**Comment by shahmishal** [comment]
Can you please xfail it in baseline?

**Comment by clackary** [comment]
@swift-ci test

**Pull request merged**